### PR TITLE
Bug 1823852: pkg/server: disable weak TLS versions

### PR DESCRIPTION
**Pull request by runcom**
Coming from an user request but it makes sense as we (OpenShift) use and control
that port.

It's not fully clear to me if we _can_ drop tls < 1.2 but I'm leaning toward so for security reasons

Signed-off-by: Antonio Murdaca <runcom@linux.com>

@cgwalters also ptal

### CODE REVIEW

File: pkg/server/api.go [hunk 1]
```diff
@@ -46,6 +47,9 @@ func (a *APIServer) Serve() {
 	mcs := &http.Server{
 		Addr:    fmt.Sprintf(":%v", a.port),
 		Handler: a.handler,
+		TLSConfig: &tls.Config{
```
Comment (cgwalters) [inline pkg/server/api.go:50]:
```suggestion
	// We don't want to allow 1.1 as that's old.  This was flagged in a security audit.
	TLSConfig: &tls.Config{
```

File: pkg/server/api.go [hunk 2]
```diff
@@ -46,6 +47,9 @@ func (a *APIServer) Serve() {
...
+			MinVersion: tls.VersionTLS12,
```
Comment (cgwalters) [inline pkg/server/api.go:51]: So 4 years from now, when the world has moved past TLS 1.2 and  Golang 1.X bumps the minimum to TLS 1.3...we'll start *downgrading* things here.  How do we avoid that?

Maybe something like having a `ClusterTlsMinVersion` constant in openshift/api?

Comment (runcom) [reply pkg/server/api.go:51]: That will impact openshift as a whole I guess but it sounds like future proof enough to me - there are other components running a server (right?)

Comment (runcom) [reply pkg/server/api.go:52]: I'll create an issue for this so we don't forget

**Comment by runcom** [comment]
This is good to go now, ptal

File: pkg/server/api.go [hunk 3]
```diff
@@ -46,6 +47,10 @@ func (a *APIServer) Serve() {
 	mcs := &http.Server{
 		Addr:    fmt.Sprintf(":%v", a.port),
 		Handler: a.handler,
+	// We don't want to allow 1.1 as that's old.  This was flagged in a security audit.
```
Comment (sinnykumari) [inline pkg/server/api.go:50]: gofmt is complaining, perhaps indentation issue. Otherwise looks sane.

Comment (runcom) [reply pkg/server/api.go:50]: fixed thanks!

**Comment by cgwalters** [comment]
> It's not fully clear to me if we can drop tls < 1.2 but I'm leaning toward so for security reasons

The only thing hitting this endpoint now should be CoreOS; so the minimum version here is really "maximum TLS supported by CoreOS". I suspect in fact we could bump it up to 1.3.

Although, I guess more precisely it's "maximum TLS supported by RHCOS 4.1...since we don't [update bootimages yet](https://github.com/openshift/enhancements/pull/201).

**Comment by runcom** [comment]
> The only thing hitting this endpoint now should be CoreOS; so the minimum version here is really "maximum TLS supported by CoreOS". I suspect in fact we could bump it up to 1.3.
> 
> Although, I guess more precisely it's "maximum TLS supported by RHCOS 4.1...since we don't [update bootimages yet](https://github.com/openshift/enhancements/pull/201).

awesome, that was my impression as well: since we control the other side of the connection, we can safely assume those are going to be the only proto to be used.

Anyway, since TLS 1.2 doesn't seem to be an issue, what if we stick with that? or do you feel strong about moving to 1.3?

**Comment by runcom** [comment]
from all the e2e tests failing tho, I'm not sure if we can do this lol will keep checking.

**Comment by sinnykumari** [comment]
/approve
will defer final review to @cgwalters

**Comment by sinnykumari** [comment]
we can fix in follow-up pr in case we can move to TLS 1.3. Let's go with 1.2 for now.
/lgtm

**Comment by kikisdeliveryservice** [comment]
PR seems to be hitting a lot of prometheus related erors in e2e-aws.. are these flakes/known bugs?

**Pull request merged**